wagtail/admin/api/serializers.py

```diff
@@ -76,8 +76,48 @@ def to_representation(self, page):
         ])
 
 
+class PageAncestorsField(Field):
+    """
+    Serializes the page's ancestry.
+
+    Example:
+    "ancestry": [
+        {
+            "id": 1,
+            "meta": {
+                "type": "wagtailcore.Page",
+                "detail_url": "/api/v1/pages/1/"
+            },
+            "title": "Root"
+        },
+        {
+            "id": 2,
+            "meta": {
+                "type": "home.HomePage",
+                "detail_url": "/api/v1/pages/2/"
+            },
+            "title": "Home"
+        }
+    ]
+    """
+    def get_attribute(self, instance):
+        return instance
+
+    def serialize_page(self, page):
+        serializer_class = get_serializer_class(page.__class__, ['id', 'type', 'detail_url', 'html_url', 'title'], meta_fields=['type', 'detail_url', 'html_url'], base=PageSerializer)
```

Comment:
Is it likely that this field will ever be serialising anything other than a vanilla `Page` object here? If not, I wonder if it might be better to define a single serializer class and reuse here, rather than creating a new class each time.

Also, since this is the Admin API, should we be aiming to show `draft_title` for pages rather than `title`? I don't know how widely that's enforced, but I think that's what the non-react explorer displays.

Reply:
Thanks for the review, I've addressed both of your comments in the commits below.